plugins/woocommerce/client/blocks/assets/js/base/components/cart-checkout/shipping-rates-control-package/package-rates.tsx

```diff
@@ -67,6 +56,17 @@ const PackageRates = ( {
 		// eslint-disable-next-line react-hooks/exhaustive-deps
 	}, [] );
 
+	// Update the selected option if cart state changes in the data store.
+	useEffect( () => {
+		if (
+			selectedRateId &&
+			selectedRateId !== previousSelectedRateId &&
+			selectedRateId !== selectedOption
+		) {
+			setSelectedOption( selectedRateId );
+		}
+	}, [ selectedRateId, selectedOption, previousSelectedRateId ] );
+
```

Comment:
Moved it for the sake of clarity. It works better in my head to have `useEffect` that runs only on mount to be above this one.